### PR TITLE
Allow to specify the table name dynamically for `ActiveRecord::class`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/ActiveRecordInterface.php [hunk 1]
```diff
@@ -173,6 +171,24 @@ public function getPrimaryKey(bool $asArray = false): mixed;
      */
     public function getRelation(string $name, bool $throwException = true): ActiveQueryInterface|null;
 
+    /**
+     * Return the name of the table associated with this AR class.
+     *
+     * Example static method:
```
Comment (xepozz) [inline src/ActiveRecordInterface.php:177]: This is not about static method

File: tests/ActiveRecordFactoryTest.php [hunk 2]
```diff
@@ -26,13 +26,22 @@ public function testCreateAR(): void
...
         $this->assertInstanceOf(Customer::class, $customerAR);
     }
 
+    public function testCreateARWithTableName(): void
+    {
+        $customerAR = $this->arFactory->createAR(Customer::class, 'customer');
+        $tableName = $customerAR->getTableName();
+
+        $this->assertSame('customer', $tableName);
+        $this->assertInstanceOf(Customer::class, $customerAR);
```
Comment (xepozz) [inline tests/ActiveRecordFactoryTest.php:42]:
```suggestion
        $customerAR = $this->arFactory->createAR(Customer::class, 'custom_table_name');
        $tableName = $customerAR->getTableName();

        $this->assertSame('custom_table_name', $tableName);
        $this->assertInstanceOf(Customer::class, $customerAR);
```

`customer` is default table name rely on class name

**Review by xepozz — CHANGES_REQUESTED** [review]

**Review by xepozz — APPROVED** [review]

File: src/ActiveQuery.php [hunk 3]
```diff
@@ -1187,7 +1188,7 @@ public function getARInstance(): ActiveRecordInterface
      */
     public function getARInstanceFactory(): ActiveRecordInterface
     {
-        return $this->arFactory->createAR($this->arClass, $this->db);
+        return $this->arFactory->createAR($this->arClass, $this->tableName, $this->db);
```
Comment (darkdef) [inline src/ActiveQuery.php:1191]: May be tableName as last parameter?

**Comment by xepozz** [comment]
So let's merge it?

**Review by darkdef — APPROVED** [review]

**Pull request merged**